compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5604,6 +5625,46 @@ private BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType c
         }
     }
 
+    private BType getTypeOfTypeParameter(BType selectType, Location pos) {
+        BType referredType = Types.getReferredType(selectType);
+        if (referredType.tag == TypeTags.INTERSECTION) {
+            referredType = ((BIntersectionType) referredType).effectiveType;
+        }
+
+        if (referredType.tag == TypeTags.UNION) {
+            BUnionType unionType = (BUnionType) referredType;
+            LinkedHashSet<BType> memberTypes = new LinkedHashSet<>();
+            for (BType type : unionType.getMemberTypes()) {
+                BType mapType = getTypeOfTypeParameter(type, pos);
+                if (mapType == symTable.semanticError) {
+                    return symTable.semanticError;
+                }
+                memberTypes.add(mapType);
+            }
+            return new BUnionType(null, memberTypes, false, false);
+        } else {
+            return getQueryMapConstraintType(referredType, pos);
+        }
+    }
+
+    private BType getQueryMapConstraintType(BType type, Location pos) {
+        type = Types.getReferredType(type);
```

Comment:
Do we need this `getReferredType` call here as well?

Reply:
My bad. Will remove this. This is not required since it is already called in previous method. Thanks.